src/auth/auth-api-request.ts

```diff
@@ -217,6 +249,10 @@ function validateCreateEditRequest(request: any, uploadAccountRequest: boolean =
       delete request[key];
     }
   }
+  if (typeof request.tenantId !== 'undefined' &&
+      !validator.isNonEmptyString(request.tenantId)) {
```

Comment:
Shouldn't the check be the mismatch between tenantId in the request and the tenantId in the user?

Reply:
There probably won't be a mismatch tenant ID backend error. For UploadAccount you can always rely on backend returning error messages in the response for specific failed users. For the other endpoints like SetAccountInfo and SignupNewUsers, the backend won't be able to return the error since it only receives one tenant_id from the request as I mentioned in a separate email thread. Adding the client side check in a follow-up change sounds good to me.

Reply:
I see. Yeah I will add the mismatch check in a follow up.